test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/MaxInputBufferLengthTests.cs

```diff
@@ -0,0 +1,248 @@
+            ManualResetEvent clientFinishedSendingRequestBody)
+        {
+            var host = new WebHostBuilder()
+                .UseKestrel(options =>
+                {
+                    options.MaxInputBufferLength = maxInputBufferLength;
+                    options.UseHttps(@"TestResources/testCert.pfx", "testPassword");
+                })
+                .UseUrls("http://127.0.0.1:0/", "https://127.0.0.1:0/")
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .Configure(app => app.Run(async context =>
+                {
+                    startReadingRequestBody.WaitOne();
+
+                    var buffer = new byte[expectedBody.Length];
+                    var bytesRead = 0;
+                    while (bytesRead < buffer.Length)
+                    {
+                        bytesRead += await context.Request.Body.ReadAsync(buffer, bytesRead, buffer.Length - bytesRead);
+                    }
+
+                    clientFinishedSendingRequestBody.WaitOne();
+
+                    // Verify client didn't send extra bytes
+                    if (context.Request.Body.ReadByte() != -1)
+                    {
+                        context.Response.StatusCode = 500;
+                        await context.Response.WriteAsync("Client sent more bytes than expectedBody.Length");
+                        return;
+                    }
+
+                    // Verify bytes received match expectedBody
+                    for (int i = 0; i < expectedBody.Length; i++)
+                    {
+                        if (buffer[i] != expectedBody[i])
+                        {
+                            context.Response.StatusCode = 500;
+                            await context.Response.WriteAsync($"Bytes received do not match expectedBody at position {i}");
+                            return;
+                        }
+                    }
+
+                    await context.Response.WriteAsync($"bytesRead: {bytesRead.ToString()}");
+                }))
+                .Build();
+
+            host.Start();
+
+            return host;
+        }
+
+        private static Socket CreateSocket(int port)
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            // Timeouts large enough to prevent false positives, but small enough to fail quickly.
+            socket.SendTimeout = 10 * 1000;
+            socket.ReceiveTimeout = 10 * 1000;
+
+            socket.Connect(IPAddress.Loopback, port);
+
+            return socket;
+        }
+
+        private static void WritePostRequestHeaders(Stream stream, int? contentLength)
+        {
+            using (var writer = new StreamWriter(stream, Encoding.ASCII, bufferSize: 1024, leaveOpen: true))
+            {
+                writer.WriteLine("POST / HTTP/1.0");
```

Comment:
The reason this is failing on non-Windows is that WriteLine is writing `\n` at the end instead of `\r\n`. Change these calls to `Write` and include `\r\n` in the strings.